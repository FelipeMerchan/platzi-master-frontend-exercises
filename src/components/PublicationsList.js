import React from 'react'

import Publication from './Publication'
import './styles/PublicationsList.css'
import data from '../data.json'

function PublicationsList() {
  return (
    <section className="PublicationsList">
      <ul>
        {data.results.map(publication => (
          <li key={publication.id}>
            <Publication
              date={publication.date}
              autor={publication.autor}
              comments={publication.comments}
              title={publication.title}
              summary={publication.summary}
            />
          </li>
        ))}
      </ul>
    </section>
  )
}

export default PublicationsList