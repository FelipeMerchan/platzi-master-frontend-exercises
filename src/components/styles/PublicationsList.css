.PublicationsList {
  margin-bottom: var(--space-2);
  padding: var(--space-2);
  padding-top: 0;
  padding-bottom: 0;
  border-radius: var(--border-radius);
  color: var(--typographyColor);
  background-color: var(--surfaceColor);
}

.PublicationsList li {
  list-style-type: none;
}