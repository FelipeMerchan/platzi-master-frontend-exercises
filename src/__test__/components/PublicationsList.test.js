import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import PublicationsList from '../../components/PublicationsList'

describe('Probar el componente <PublicationsList />', () => {
  const publicationsList = shallow(<PublicationsList />)

  test('Render del componente PublicationsList', () => {
    expect(publicationsList.length).toEqual(1)
  })
  test('Render de las publicaciones', () => {
    expect(publicationsList.find('ul').children()).toHaveLength(4);
  });
  test('Render con el className PublicationsList', () => {
    expect(publicationsList.exists('.PublicationsList')).toBeTruthy()
  })
})

describe('PublicationsList Snapshot', () => {
  test('Comprobar la UI del componente PublicationsList', () => {
    const publicationsList = create(<PublicationsList />)
    expect(publicationsList.toJSON()).toMatchSnapshot()
  })
})