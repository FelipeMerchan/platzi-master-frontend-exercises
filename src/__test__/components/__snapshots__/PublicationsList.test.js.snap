// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationsList Snapshot Comprobar la UI del componente PublicationsList 1`] = `
<section
  className="PublicationsList"
>
  <ul>
    <li>
      <article
        className="Publication"
      >
        <div
          className="Publication__data"
        >
          <span>
            El 
            29 de octubre de 2012
             
            <br />
            Por 
            <span
              className="Publication__author"
            >
              Ernesto G Bustamante
            </span>
            <br />
            <small>
              3
               comentarios
            </small>
          </span>
        </div>
        <div
          className="Publication__details"
        >
          <h2
            className="Publication__title mb"
          >
            Las bondades de HTML5
          </h2>
          <p
            className="Publication__text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
          </p>
          <a
            className="Publication__link"
          >
            Leer más
          </a>
        </div>
      </article>
    </li>
    <li>
      <article
        className="Publication"
      >
        <div
          className="Publication__data"
        >
          <span>
            El 
            El 26 de octubre de 2012
             
            <br />
            Por 
            <span
              className="Publication__author"
            >
              Ernesto G Bustamante
            </span>
            <br />
            <small>
              21
               comentarios
            </small>
          </span>
        </div>
        <div
          className="Publication__details"
        >
          <h2
            className="Publication__title mb"
          >
            HTML5 en dispositivos móviles
          </h2>
          <p
            className="Publication__text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
          </p>
          <a
            className="Publication__link"
          >
            Leer más
          </a>
        </div>
      </article>
    </li>
    <li>
      <article
        className="Publication"
      >
        <div
          className="Publication__data"
        >
          <span>
            El 
            27 de octubre de 2012
             
            <br />
            Por 
            <span
              className="Publication__author"
            >
              Ernesto G Bustamante
            </span>
            <br />
            <small>
              6
               comentarios
            </small>
          </span>
        </div>
        <div
          className="Publication__details"
        >
          <h2
            className="Publication__title mb"
          >
            Internet Explorer es una basura
          </h2>
          <p
            className="Publication__text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
          </p>
          <a
            className="Publication__link"
          >
            Leer más
          </a>
        </div>
      </article>
    </li>
    <li>
      <article
        className="Publication"
      >
        <div
          className="Publication__data"
        >
          <span>
            El 
            26 de octubre de 2012
             
            <br />
            Por 
            <span
              className="Publication__author"
            >
              Ernesto G Bustamante
            </span>
            <br />
            <small>
              2
               comentarios
            </small>
          </span>
        </div>
        <div
          className="Publication__details"
        >
          <h2
            className="Publication__title mb"
          >
            Diseño web en pantallas retina
          </h2>
          <p
            className="Publication__text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
          </p>
          <a
            className="Publication__link"
          >
            Leer más
          </a>
        </div>
      </article>
    </li>
  </ul>
</section>
`;
